### PR TITLE
[NFC]Rename InstrProf::getFuncName{,orExternalSymbol} to getValueName{,orExternalSymbol}

### PR DESCRIPTION
**Pull request by mingmingl-llvm**
- This function looks up MD5ToNameMap to return a name for a given MD5. https://github.com/llvm/llvm-project/pull/66825 adds MD5 of global variable names into this map. So rename methods and update comments

### CODE REVIEW

File: llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h [hunk 1]
```diff
@@ -813,7 +813,7 @@ template <class FuncRecordTy, support::endianness Endian>
 Error getFuncNameViaRef(const FuncRecordTy *Record,
                         InstrProfSymtab &ProfileNames, StringRef &FuncName) {
   uint64_t NameRef = getFuncNameRef<FuncRecordTy, Endian>(Record);
-  FuncName = ProfileNames.getFuncName(NameRef);
+  FuncName = ProfileNames.getValueName(NameRef);
```
Comment (david-xl) [inline llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h:816]: nit: make the name clearer: getFuncOrVarName().

File: llvm/include/llvm/ProfileData/InstrProf.h [hunk 2]
```diff
@@ -510,14 +510,14 @@ class InstrProfSymtab {
...
+  inline StringRef getValueName(uint64_t ValMD5Hash);
 
-  /// Just like getFuncName, except that it will return a non-empty StringRef
-  /// if the function is external to this symbol table. All such cases
-  /// will be represented using the same StringRef value.
-  inline StringRef getFuncNameOrExternalSymbol(uint64_t FuncMD5Hash);
+  /// Just like getValueName, except that it will return a non-empty StringRef
+  /// if the value (function or global variable) is external to this symbol
+  /// table. All such cases will be represented using the same StringRef value.
+  inline StringRef getValueNameOrExternalSymbol(uint64_t ValMD5Hash);
```
Comment (david-xl) [inline llvm/include/llvm/ProfileData/InstrProf.h:520]: getFuncOrVarNameIfDefined().  Update comment to say 'External Symbol' is returned if not found.

File: llvm/include/llvm/ProfileData/InstrProf.h [hunk 3]
```diff
@@ -565,14 +565,14 @@ void InstrProfSymtab::finalizeSymtab() {
   Sorted = true;
 }
 
-StringRef InstrProfSymtab::getFuncNameOrExternalSymbol(uint64_t FuncMD5Hash) {
-  StringRef ret = getFuncName(FuncMD5Hash);
+StringRef InstrProfSymtab::getValueNameOrExternalSymbol(uint64_t FuncMD5Hash) {
```
Comment (david-xl) [inline llvm/include/llvm/ProfileData/InstrProf.h:568]: FuncMD5Hash -- MD5Hash

File: llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h (hunk 1 above)
Comment (minglotus-6) [reply llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h]: done.

File: llvm/include/llvm/ProfileData/InstrProf.h (hunk 2 above)
Comment (minglotus-6) [reply llvm/include/llvm/ProfileData/InstrProf.h]: done.

File: llvm/include/llvm/ProfileData/InstrProf.h (hunk 3 above)
Comment (minglotus-6) [reply llvm/include/llvm/ProfileData/InstrProf.h]: done.

**Review by david-xl — APPROVED** [review]

**Pull request merged**